migrations/2023-10-13-080226_add-features-limit/up.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE crates ADD max_features SMALLINT;
```

Comment:
This maxes us out at 32,767 possible features for a crate version ever. I'm personally OK with this, but are we sure there isn't a case where this could bite us later?

(I mean, I guess practically we could just change it if and when we need to.)

Reply:
> are we sure there isn't a case where this could bite us later?

there are probably use cases that could use more features than that, but given the current technological constraints allowing that many features for any crate has significant consequences. before those constraints are resolved I wouldn't be comfortable allowing crates with more than 32k features, and afterwards we probably won't even need the limit anymore.